### ISSUE TITLE
FE task to create a new button "Uninstall" for Pipedrive connector

### ISSUE THREAD

**Issue by igrankova** [airbytehq/airbyte#8765]
## Tell us about the problem you're trying to solve
We need to implement the "Uninstall" button for the source connector Pipedrive. It is a must condition to submit the Public App.

## Describe the solution you’d like

1. On the connection Settings screen add a button "Uninstall".
2. The button should be enabled by default.
3. On button press implement Uninstallation flow (should be implemented in the other task) - #8766
4. See the mockup below.
![Uninstall](https://user-images.githubusercontent.com/87977540/145980605-35d09003-7487-49ad-bbfd-89a8eccc9a5c.png)

**Comment by igrankova** [comment]
@jacob-a6, @Jamakase, @sherifnada, could you please review the tasks and the mockup I made? We received a requirement to implement Uninstallation flow from the Pipedrive support as a condition to make our OAuth app public.

**Comment by Jamakase** [comment]
@igrankova How does uninstall differ from deleting the destination? Do we just revoke access in this case or we also delete the destination entity? Is it possible to revoke permissions but keep the destination enabled? I think no. Doesn't it mean then that deleting the destination itself should revoke all permissions?

Also 1 more thing - what if numerous destinations are used for oauth for pipedrive and same account - will they be assumed as different applications and user will need to revoke them 1 by one, or revoking 1 destination will make all other connectors non working too?

**Comment by igrankova** [comment]
@Jamakase, I apologize for the wrong pic. The button should be on the connection Settings screen and it should delete the installed app from the Pipedrive account and revoke all connections and access to the Pipedrive account of the current user. "if a user uninstalls it (app) and your server should also know it so that it can stop sending requests for that specific user." - https://pipedrive.readme.io/docs/app-uninstallation

**Comment by timroes** [comment]
Is this still a valid issue we're looking into implementing?

If so, given that the app is linked to the connector (not a connection) I think this button shouldbe part of the connector setting and not a connection setting page. Also should this connector maybe automatically remove the application if the last connector of this type is deleted?

**Comment by sherifnada** [comment]
I don't think this is something we're prioritizing for now. @misteryeo this was required for making Pipedrive oauth available in cloud. I'll leave it up to you to figure out whether to close it or move it to the appropriate queue. 

FWIW I think when we do get to this, it'll need to happen on the platform as a "post-hook" of sorts. There are probably other integrations with similar needs.

**Comment by timroes** [comment]
Closing for now as not planned. Happy to reopen in case we're planning to pick this up.

**Comment by YowanR** [comment]
@timroes This is potential something that we'd like to bring back this quarter. Could you provide some context of LoE for this work, please?

**Comment by timroes** [comment]
The most work here is on the BE side (building the API to uninstall connectors, or extending the delete API to clear oauth data, adding information into connector definitions on whether that feature is supported, etc.). The FE work of this should be fairly trivial (1 or 2 weeks), once all the APIs are in place. I'm not even sure if there's any FE work at all in this, since I think deleting a connector should just do those tasks automatically.

I think the following things need to be clarified for this to be worked on:

* Should this be a manual action or just automation during deleting a connector. Since multiple connectors could use the same account to talk to Pipedrive I assume if we do it automatically we'll need to check that we only disconnect after the last connector using this account has been deleted. If it's a manual action, we have the problem that users might delete all connectors of this type without disconnecting, so we might need a separate place in settings to disconnect (which is likely users will anyway not find). Also a manual action would still need to check somehow that you're not disconnecting while there are still other connectors using the same account.
* I think we should not call it "uninstall" since that sounds very confusing. It sounds more like you'd uninstall the Pipedrive connector for Airbyte, but we're just simply disconnecting from Pipedrive OAuth again.

Given that this requires extension on the connector definition/spec side, platform APIs and FE changes, I think you should provide a tech spec for this task, before starting to work on it, so we can clarify aboves question and other questions that might come up during this.

**Comment by YowanR** [comment]
Thanks for explaining, @timroes! Super helpful Do you know who owns the BE for this project? Would that be under Malik?
cc @misteryeo as FYI here

**Comment by timroes** [comment]
I think that would be likely the Platform Workflow team with @salima-airbyte, since they usually own the fundamental APIs that are called.

**Comment by YowanR** [comment]
Cool, thank you! @salima-airbyte would love to hear your thoughts on this. Ideally, I'd like to get some understanding of the LoE and if we could potentially fit this in this quarter?

**Comment by misteryeo** [comment]
cc: @andyjih for viz

**Comment by YowanR** [comment]
Re-opening this ticket as we are actively trying to scope it

**Comment by andyjih** [comment]
Just to jump in, I think it's pretty unlikely that the workflow backend work would happen this quarter. I think it'd still be useful to have a team member scope it out though so we can see what the lift will be in the future. @salima-airbyte

**Comment by gosusnp** [comment]
I agree with previous comments, the `uninstall` term feels a bit misleading, instinctively, I expected this to be about deleting the pipedrive connector while this seems to be about revoking tokens.

Looking at the mockup, this `uninstall` is done configured at connection level, however, the way our data model is currently setup, this token is store for a `(workspace, actor definition)`, which means that this would impact all the connections of the workspace using pipedrive. If this is not a desirable behavior, we may need to look at correcting the data model first to have tokens at connection level rather than workspace.

On the BE, adding an oauth token revokation endpoint should remain fairly low. Most of the complexity would come from how we want to implement the revokation calls. Looking at the [pipedrive doc](https://pipedrive.readme.io/docs/app-uninstallation), this might fit within a more generic pattern for OAuth token revoke since I'd expect other providers to also have this.

**Comment by salima-airbyte** [comment]
@YowanR per our discussion, please connect with @evantahler. @gosusnp researched the ticket and put up this [doc](https://docs.google.com/document/d/1Ix4GiAfS3TYkjHLvGLbkZMBqzA0e0jRXcMHblo14R9w/edit#). Please let us know if there is any question.

**Comment by bazarnov** [comment]
Status note (2026-09-15): the source-pipedrive OAuth work in progress (https://github.com/airbytehq/airbyte/pull/85773) uses a private, unlisted Pipedrive app installed through the OAuth consent flow, which Pipedrive allows without a Marketplace listing. The uninstall flow described here is only required to submit a Public Marketplace app, so it does not block that work; it becomes relevant again only if the app is later submitted as Public.
